mopro-ffi/src/lib.rs

```diff
@@ -67,6 +72,30 @@ macro_rules! halo2_app {
     };
 }
 
+#[cfg(not(feature = "nova_scotia"))]
+#[macro_export]
+macro_rules! nova_scotia_app {
+    () => {
+        fn generate_nova_scotia_proof(
+            r1cs_path: String,
+            cpp_bin_or_wasm_path: String,
+            private_inputs: Vec<HashMap<String, Value>>,
+            start_public_input: Vec<F<G1>>,
+        ) -> Result<GenerateProofResult, MoproError> {
+            panic!("Nova Scotia is not enabled in this build. Please pass `nova-scotia` feature to `mopro-ffi` to enable Nova Scotia.")
+        }
+
+        fn verify_nova_scotia_proof(
+            recursive_snark: RecursiveSNARK<G1, G2, C1<G1>, C2<G2>>,
+            pp: &PublicParams<G1, G2, C1<G1>, C2<G2>>,
+            iteration_count: usize,
+            start_public_input: &[E1::Scalar],
```

Comment:
I found that there are some problems with these data types
it causes the CI error (refer to: https://github.com/zkmopro/mopro/actions/runs/11051448766/job/30701493136)

in mopro-ffi tests you import these dependencies [here](https://github.com/csiejimmyliu/mopro/blob/a7bbad4d781effe38ef2e0551ac4db2532f4d3c5/mopro-ffi/src/nova_scotia/mod.rs#L92)
but like `test-e2e` doesn't have these dependencies so it says like
```sh
use of undeclared type `E1`
```

The possible solution might be: 
1. Use serialization like circom [serialization.rs](https://github.com/zkmopro/mopro/blob/main/mopro-ffi/src/circom/serialization.rs). 
   To make the type the native rust types: `Vec`, `String`, `u8`, `std::collections::HashMap`,...
2. or import the types in `src/lib.rs` (not recommended, but could be fast to solve)
   Since we will convert the type in `mopro.udl`, and the built-in types are: https://mozilla.github.io/uniffi-rs/0.27/udl/builtin_types.html